src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp

```diff
@@ -105,7 +120,23 @@ __device__ auto block_reduce(index idx, Op op, T init, Index n, F f)
 #endif
     using type = decltype(index::invoke_loop(f, 0, _c<0>));
     __shared__ type buffer[idx.max_nlocal() / lanes_per_thread];
+    // Hack: Note when in int32_t and lowest/highest op, I need to overrid the init value
     type x = init;
+    type y = init;
+    if constexpr(is_same<type, int32_t>{})
+    {
+        if constexpr(is_same<T, lowest>{})
+        {
+            x = INT_MIN;
+            y = INT_MIN;
```

Comment:
If this is needed, then I think we might have a bug in `numeric_lowest`.

Reply:
I stared at `numeric_lowest` like ten times, and it looks correct. Just don't know how to debug that...

Reply:
`numeric_lowest` does `-numeric_max<T>() - 1`, I dont think that will overflow, but maybe an implicit conversion is causing an issue.

Reply:
FYI, printing out the x as-is in `%d` (without the hack) will give me a mysterious `-2069889000`. Does not occur to be a number that can overflow to.

Reply:
But that is a wrong value, shouldn't it be `-2147483648`?